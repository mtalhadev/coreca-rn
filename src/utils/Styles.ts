import { StyleSheet } from 'react-native'
import { THEME_COLORS } from './Constants'

export const FontStyle = {
    thin: 'NotoSansJP_100Thin',
    light: 'NotoSansJP_300Light',
    regular: 'NotoSansJP_400Regular',
    medium: 'NotoSansJP_500Medium',
    bold: 'NotoSansJP_700Bold',
    black: 'NotoSansJP_900Black',
}

export const GlobalStyles = StyleSheet.create({
    normalText: {
        fontSize: 14,
        fontFamily: FontStyle.regular,
        lineHeight: 16,
    },
    headerText: {
        fontSize: 20,
        fontFamily: FontStyle.bold,
        lineHeight: 22,
    },
    mediumText: {
        fontSize: 13,
        fontFamily: FontStyle.medium,
        lineHeight: 15,
    },
    boldText: {
        fontSize: 14,
        fontFamily: FontStyle.bold,
        lineHeight: 16,
    },
    smallText: {
        fontSize: 12,
        fontFamily: FontStyle.regular,
        lineHeight: 14,
    },
    smallGrayText: {
        fontSize: 12,
        fontFamily: FontStyle.regular,
        lineHeight: 14,
        color: THEME_COLORS.OTHERS.GRAY,
    },
})

export type ColorStyle = {
    mainColor: string
    subColor: string
    deepColor: string
    lightColor: string
    textColor: string
    deepTextColor: string
    highLightColor: string
    focusColor: string
    highLightTextColor: string
    menuHighlightColor: string
    anotherColorStyle: ColorStyle | undefined
}

export const BlueColor: ColorStyle = {
    mainColor: THEME_COLORS.BLUE.MIDDLE,
    subColor: THEME_COLORS.BLUE.MIDDLE_DEEP,
    deepColor: THEME_COLORS.BLUE.SUPER_DEEP,
    lightColor: THEME_COLORS.BLUE.SUPER_LIGHT,
    textColor: '#fff',
    deepTextColor: THEME_COLORS.BLUE.MIDDLE_DEEP,
    highLightColor: THEME_COLORS.BLUE.HIGH_LIGHT,
    focusColor: THEME_COLORS.BLUE.SUPER_DEEP,
    highLightTextColor: THEME_COLORS.BLUE.LIGHT,
    menuHighlightColor: THEME_COLORS.GREEN.LIGHT,
    anotherColorStyle: undefined,
}

export const GreenColor: ColorStyle = {
    mainColor: THEME_COLORS.GREEN.LIGHT,
    subColor: THEME_COLORS.GREEN.MIDDLE,
    deepColor: THEME_COLORS.GREEN.SUPER_DEEP,
    lightColor: THEME_COLORS.GREEN.SUPER_LIGHT,
    textColor: '#000',
    deepTextColor: THEME_COLORS.GREEN.DEEP,
    highLightColor: '#fff',
    focusColor: '#fff',
    highLightTextColor: THEME_COLORS.GREEN.MIDDLE,
    menuHighlightColor: THEME_COLORS.BLUE.LIGHT,
    anotherColorStyle: BlueColor,
}
BlueColor.anotherColorStyle = GreenColor
