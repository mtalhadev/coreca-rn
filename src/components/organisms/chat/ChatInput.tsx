import React, { useCallback, useEffect, useRef, useState } from 'react'
import { Image, Linking, Pressable, Text, TextInput, View, ViewStyle } from 'react-native'

import { FontStyle, GlobalStyles } from '../../../utils/Styles'
import _, { trim } from 'lodash'
import { IPHONEX_BOTTOM_HEIGHT, THEME_COLORS, WINDOW_WIDTH } from '../../../utils/Constants'
import { MessageCLType } from '../../../models/message/Message'
import { useTextTranslation } from '../../../fooks/useTextTranslation'
import { Icon } from '../../atoms/Icon'
import { pickImage, resizeImage } from '../../../utils/Utils'
import { _uploadImageAndGetUrl } from '../../../services/firebase/StorageService'
import { ThreadLogType } from '../../../models/threadLog/ThreadLog'
import { MentionPopup } from './MentionPopup'
import { RoomUserType } from '../../../models/roomUser/RoomUser'
import ParsedText from 'react-native-parsed-text'
import VectorIcon from '../../atoms/VectorIcon'
import { ImageInfo } from 'expo-image-picker/build/ImagePicker.types'
import { isIphoneX } from 'react-native-iphone-screen-helper'
import { navigate } from '../../../screens/RootNavigation'

export type ChatInputProps = {
    reply?: MessageCLType
    threadReply?: ThreadLogType
    visible?: boolean
    roomUsers: RoomUserType[]
    style?: ViewStyle
    showAtIcon?: boolean
    onPost?: (message: string) => void
    onPostAttachment?: (image: ImageInfo) => void
    onCloseReply?: () => void
    onCloseThreadReply?: () => void
    onNotePress?: () => void
    onTodoPress?: () => void
}

const viewRef = React.createRef<View>()

export const ChatInput = React.memo((props: Partial<ChatInputProps>) => {
    let { reply, threadReply, visible, style, roomUsers, showAtIcon, onPost, onPostAttachment, onCloseReply, onCloseThreadReply, onNotePress, onTodoPress } = props
    const { t } = useTextTranslation()

    const [textInput, setTextInput] = useState<string>('')
    const [mentionInput, setMentionInput] = useState<string>('')
    const [showPopover, setShowPopover] = useState<boolean>(false)
    const [isMessageStarted, setIsMessageStarted] = useState(true)
    const [members, setMembers] = useState<RoomUserType[]>(roomUsers || [])
    const [mentionUSer, setMentionUSer] = useState<RoomUserType>()
    const [imageInfo, setImageInfo] = useState<ImageInfo>()

    // console.log('mentionUSer',mentionUSer);

    useEffect(() => {
        if (roomUsers?.length) {
            setMembers(roomUsers)
        }
    }, [roomUsers])

    useEffect(() => {
        if (mentionUSer) {
            const index = textInput.lastIndexOf('@')
            setTextInput(textInput.substring(0, index).concat(`@${mentionUSer.worker?.name} `))
        }
    }, [mentionUSer])

    visible = visible ?? true
    showAtIcon = showAtIcon ?? true

    const checkEmpty = (text: string | undefined): boolean => {
        if (text == undefined) {
            return true
        }
        if (trim(text) == '') {
            return true
        }
        return false
    }

    const _onChangeText = (text: string) => {
        setTextInput(text)
    }
    const _onKeyPress = (key: string) => {
        if (key == '@') {
            setShowPopover(true)
            setIsMessageStarted(false)
        } else if (key == ' ') {
            setShowPopover(false)
            setIsMessageStarted(true)
            setMentionInput('')
        } else {
            if (!isMessageStarted) {
                if (key !== 'Backspace') setMentionInput(mentionInput + key)
                else setMentionInput(mentionInput.slice(0, -1) || '')
            }
        }
        // console.log('key pressed: ', key)
    }
    // console.log('=== mentionInput: ', mentionInput);

    const _handleUrlPress = async (url: string) => {
        if (await Linking.canOpenURL(url)) await Linking.openURL(url)
    }
    const _handleEmailPress = (email: string) => {
        Linking.openURL(`mailto:${email}`)
    }
    const _handlePhonePress = (phone: string) => {
        Linking.openURL(`sms:${phone}`)
    }
    const _handleMentionNamePress = (mentionName: string) => {
        setMentionUSer(roomUsers?.find((item) => item.worker?.name && mentionName.includes(item.worker?.name)))
        setShowPopover(true)
    }

    const _onSelectUser = (workerId: string) => {
        setShowPopover(false)
        console.log('workerId: ', workerId)
        const roomUser = roomUsers?.find((item) => item.worker?.workerId === workerId)
        setMentionUSer(roomUser)
    }

    const filtered = members.filter((item) => item.worker?.name?.startsWith(mentionInput))

    return (
        <View
            ref={viewRef}
            style={[
                {
                    flexDirection: 'column',
                    display: visible ? 'flex' : 'none',
                    padding: 10,
                    paddingBottom: isIphoneX() ? IPHONEX_BOTTOM_HEIGHT : 0,
                    backgroundColor: '#fff',
                    // borderColor: THEME_COLORS.OTHERS.GRAY,
                    // borderTopWidth: 0.5
                },
                style,
            ]}>
            {reply && (
                <View
                    style={{
                        flexDirection: 'row',
                    }}>
                    <View style={{ flex: 1 }}>
                        <Text
                            style={{
                                color: THEME_COLORS.BLUE.MIDDLE,
                            }}>
                            Re : {reply?.worker?.name}
                        </Text>
                        <Text
                            style={{
                                color: THEME_COLORS.OTHERS.GRAY,
                            }}
                            numberOfLines={1}
                            ellipsizeMode={'tail'}>
                            {reply?.message}
                        </Text>
                    </View>
                    <Pressable
                        onPress={() => {
                            if (onCloseReply) {
                                onCloseReply()
                            }
                        }}>
                        <Icon name="close-reply" fill={THEME_COLORS.BLUE.HIGH_LIGHT} />
                    </Pressable>
                </View>
            )}
            {threadReply && (
                <View
                    style={{
                        borderTopColor: THEME_COLORS.BLUE.MIDDLE,
                        borderTopWidth: 1,
                        flexDirection: 'row',
                        paddingVertical: 8,
                        paddingHorizontal: 8,
                    }}>
                    <View style={{ flex: 1 }}>
                        <Text
                            style={{
                                fontFamily: FontStyle.bold,
                            }}
                            numberOfLines={1}
                            ellipsizeMode={'tail'}>
                            {threadReply.room?.name ? threadReply.room?.name + 'へ投稿' : ''}
                        </Text>
                    </View>
                    <Pressable
                        onPress={() => {
                            if (onCloseThreadReply) {
                                onCloseThreadReply()
                            }
                        }}>
                        <Icon name="close-reply" fill={THEME_COLORS.BLUE.HIGH_LIGHT} />
                    </Pressable>
                </View>
            )}
            <View
                style={{
                    flexDirection: 'column',
                }}>
                <View
                    style={{
                        flexDirection: 'row',
                    }}>
                    <TextInput
                        multiline={true}
                        onChangeText={_onChangeText}
                        onKeyPress={(e) => _onKeyPress(e.nativeEvent.key)}
                        style={{
                            borderColor: THEME_COLORS.OTHERS.GRAY,
                            borderWidth: 0.5,
                            borderRadius: 15,
                            marginHorizontal: 0,
                            flex: 1,
                            paddingTop: 9,
                            paddingHorizontal: 10,
                            ...GlobalStyles.normalText,
                        }}>
                        <ParsedText
                            style={{
                                fontFamily: FontStyle.regular,
                                fontSize: 12,
                            }}
                            parse={[
                                {
                                    type: 'url',
                                    style: {
                                        color: THEME_COLORS.BLUE.MIDDLE_DEEP,
                                    },
                                    onPress: _handleUrlPress,
                                },
                                {
                                    type: 'phone',
                                    style: {
                                        color: THEME_COLORS.BLUE.MIDDLE_DEEP,
                                    },
                                    onPress: _handlePhonePress,
                                },
                                {
                                    type: 'email',
                                    style: {
                                        color: THEME_COLORS.BLUE.MIDDLE_DEEP,
                                    },
                                    onPress: _handleEmailPress,
                                },
                                {
                                    pattern: /@(\w+)|@(従業員\d{5})/,
                                    style: {
                                        color: THEME_COLORS.BLUE.MIDDLE_DEEP,
                                        fontFamily: FontStyle.bold,
                                    },
                                    onPress: _handleMentionNamePress,
                                },
                            ]}
                            childrenProps={{ allowFontScaling: false }}>
                            {textInput}
                        </ParsedText>
                    </TextInput>

                    <Pressable
                        onPress={async () => {
                            if (onPost && !checkEmpty(textInput)) {
                                onPost(textInput.trim())
                                setTextInput('')
                            } else if (imageInfo && imageInfo.uri) {
                                if (onPostAttachment) {
                                    onPostAttachment(imageInfo)
                                }
                                setImageInfo(undefined)
                            }
                        }}
                        style={({ pressed }) => ({
                            opacity: pressed ? 0.7 : 1,
                        })}>
                        <View
                            style={{
                                width: 35,
                                height: 35,
                                marginLeft: 8,
                                backgroundColor: THEME_COLORS.BLUE.MIDDLE,
                                borderRadius: 25,
                                justifyContent: 'center',
                                alignItems: 'center',
                            }}>
                            <VectorIcon.FontAwesome name="send" size={18} color="#FFF" style={{ marginRight: 2 }} />
                        </View>
                    </Pressable>

                    {Boolean(showAtIcon) === true && <MentionPopup data={filtered.length ? filtered : members} visible={showPopover} selected={mentionUSer} onSelectUser={_onSelectUser} />}
                </View>

                {imageInfo && (
                    <View
                        style={{
                            width: 110,
                            height: 115,
                            marginTop: 6,
                            alignSelf: 'center',
                        }}>
                        <Image
                            style={{
                                marginTop: 6,
                                width: 100,
                                height: 100,
                                borderRadius: 10,
                                resizeMode: 'contain',
                                backgroundColor: THEME_COLORS.OTHERS.SUPER_LIGHT_GRAY,
                            }}
                            source={{ uri: imageInfo.uri }}
                        />
                        <Pressable
                            onPress={() => {
                                setImageInfo(undefined)
                            }}
                            style={{
                                width: 20,
                                height: 20,
                                borderRadius: 20,
                                position: 'absolute',
                                top: 0,
                                right: 0,
                                backgroundColor: THEME_COLORS.OTHERS.LIGHT_GRAY,
                                justifyContent: 'center',
                                alignItems: 'center',
                            }}>
                            <Icon name="close" width={8} height={8} fill={THEME_COLORS.OTHERS.BACKGROUND} />
                        </Pressable>
                    </View>
                )}

                <View
                    style={{
                        flexDirection: 'row',
                        justifyContent: 'flex-start',
                        marginBottom: 5,
                    }}>
                    {Boolean(showAtIcon) === true && (
                        <Pressable
                            onPress={() => {
                                if (onCloseReply) {
                                    onCloseReply()
                                }
                                setShowPopover(!showPopover)
                            }}
                            style={{
                                marginTop: 5,
                                marginRight: 15,
                                marginLeft: 5,
                            }}>
                            <Text
                                style={{
                                    fontFamily: FontStyle.bold,
                                    fontSize: 20,
                                    lineHeight: 26,
                                    color: THEME_COLORS.OTHERS.GRAY,
                                }}>
                                @
                            </Text>
                        </Pressable>
                    )}
                    {Boolean(onPostAttachment) === true && (
                        <Pressable
                            onPress={async () => {
                                const image = await pickImage(false)
                                setImageInfo(image)
                            }}
                            style={{
                                marginTop: 10,
                            }}>
                            <Icon name="attachment" width={20} height={20} />
                        </Pressable>
                    )}
                    {Boolean(onNotePress) === true && (
                        <Pressable
                            onPress={() => {
                                if (onNotePress) onNotePress()
                            }}
                            style={{
                                marginTop: 10,
                                marginLeft: 16,
                            }}>
                            <VectorIcon.FontAwesome name="file-text-o" size={20} color={THEME_COLORS.OTHERS.GRAY} />
                        </Pressable>
                    )}
                    {Boolean(onTodoPress) === true && (
                        <Pressable
                            onPress={() => {
                                if (onTodoPress) onTodoPress()
                            }}
                            style={{
                                marginTop: 10,
                                marginLeft: 16,
                            }}>
                            <VectorIcon.Octicon name="tasklist" size={20} color={THEME_COLORS.OTHERS.GRAY} />
                        </Pressable>
                    )}
                </View>
            </View>
        </View>
    )
})

export default ChatInput
