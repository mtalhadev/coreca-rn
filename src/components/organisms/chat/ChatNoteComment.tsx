import React from 'react'
import { Image, Pressable, Text, View, ViewStyle } from 'react-native'

import * as Clipboard from 'expo-clipboard'
import { FontStyle } from '../../../utils/Styles'

import { THEME_COLORS, WINDOW_WIDTH } from '../../../utils/Constants'
import { ImageIcon } from '../ImageIcon'

import {
    Menu, MenuOption, MenuOptions, MenuTrigger, renderers
} from 'react-native-popup-menu'
import { useTextTranslation } from '../../../fooks/useTextTranslation'
import { NoteCommentCLType } from '../../../models/noteComment/NoteComment'
import { CustomDate, timeText } from '../../../models/_others/CustomDate'
import { Icon } from '../../atoms/Icon'
import VectorIcon from '../../atoms/VectorIcon'

export type ChatNoteCommentProps = {
    comment: NoteCommentCLType
    style?: ViewStyle
    onAttachmentPress?: (attachment: string) => void 

}

export const ChatNoteComment = React.memo((props: Partial<ChatNoteCommentProps>) => {
    const {
        comment,
        style,
        onAttachmentPress,
     } = props;

    const iconSize = 25
    const _imageUri = (iconSize <= 30 ? comment?.worker?.xsImageUrl : iconSize <= 50 ? comment?.worker?.sImageUrl : comment?.worker?.imageUrl) ?? comment?.worker?.imageUrl
    const { t } = useTextTranslation()
   
    return (
        <View
            style={[
                {
                    flexDirection: 'column',
                    borderBottomColor: THEME_COLORS.OTHERS.LIGHT_GRAY,
                    borderBottomWidth: 0.3,
                },
                style,
            ]}>
            <Pressable
                style={{
                    flexDirection: 'row',
                    marginVertical: 8,
                }}
            >

                <ImageIcon 
                    imageColorHue={comment?.worker?.imageColorHue} 
                    imageUri={_imageUri} 
                    type={'worker'} 
                    size={iconSize} 
                    style={{ marginTop: 5 }}
                    />

                <View
                    style={{
                        marginLeft: 10,
                        justifyContent: 'center',
                    }}>
                    <View
                        style={{
                            flexDirection: 'column',
                            alignItems: 'flex-start',
                            paddingRight: 6,
                        }}>
                        <Text
                            numberOfLines={1}
                            ellipsizeMode={'middle'}
                            style={{
                                fontFamily: FontStyle.bold,
                                fontSize: 12,
                                lineHeight: 14,
                                width: WINDOW_WIDTH - 110,
                            }}>
                            {comment?.worker?.name}
                        </Text>
                        {!!comment?.message?.length && 
                        <Text
                            ellipsizeMode='middle'
                            style={{
                                fontFamily: FontStyle.regular,
                                marginTop: 6,
                                fontSize: 12,
                                lineHeight: 16,
                                width: WINDOW_WIDTH - 110,
                            }}
                            >
                                {comment?.message}
                        </Text>
                        }
                        {(comment?.xsAttachmentUrl) && 
                        <Pressable 
                            onPress={() => {
                                if(onAttachmentPress)  
                                    onAttachmentPress(comment.attachmentUrl ?? 'https://no-url')
                            }} 
                        >
                            <Image
                                style={{
                                    marginTop: 6,
                                    width: 160,
                                    height: 120,
                                    resizeMode: 'contain'
                                }} 
                                source={{uri: comment?.xsAttachmentUrl ?? 'https://no-url'}}
                                />
                        </Pressable>
                        }
                    </View>
                </View>
                <View
                    style={{
                        alignItems: 'flex-end',
                    }}
                >
                    <View
                        style={{
                            flexDirection: 'column',
                            alignItems: 'flex-end'
                        }}
                        
                    >
                        <Text
                            style={{
                                color: THEME_COLORS.OTHERS.GRAY,
                                fontSize: 10,
                                // backgroundColor: '#ddd',
                                marginTop: 20
                            }}
                        >
                            {timeText(comment?.createdAt as CustomDate).substring(0, 5)}
                        </Text>
                        {/* <Menu style={{
                            marginTop: 10
                        }} renderer={renderers.Popover}>
                            <MenuTrigger 
                                customStyles={{
                                    triggerWrapper: {
                                        top: -42,
                                        // right: 10
                                    },
                                }}
                                >
                                <VectorIcon.Entypo name='dots-three-horizontal' color={THEME_COLORS.OTHERS.GRAY} size={13}/>
                            </MenuTrigger>
                            <MenuOptions
                                optionsContainerStyle={{
                                    top: -48,
                                    right: 25
                                }}
                                customStyles={{
                                    optionsContainer: {
                                        justifyContent: 'center',
                                        borderRadius: 5,

                                    },
                                    optionsWrapper: {
                                        flexDirection: 'row',
                                        borderRadius: 8,
                                        backgroundColor: 'white',
                                        justifyContent: 'center',
                                        alignItems: "center",                   
                                    },
                                }}
                            >
                                <MenuOption onSelect={() => Clipboard.setStringAsync(comment?.message as string)}>
                                    <Icon name='copy' fill={'#fff'} />
                                </MenuOption>
                                <MenuOption onSelect={() => onDelete && onDelete(comment?.noteCommentId ?? 'no-id')}>
                                    <VectorIcon.Feather name='trash-2' color={THEME_COLORS.BLUE.MIDDLE} size={17} />
                                </MenuOption>
                            </MenuOptions>
                        </Menu> */}

                    </View>
                </View>
            </Pressable>
            
        </View>

    )
})
