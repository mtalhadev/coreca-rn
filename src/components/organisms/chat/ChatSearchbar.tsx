import React, { useState } from 'react';
import { View,StyleSheet, TextInput } from 'react-native';
import { SCREEN_WIDTH, THEME_COLORS } from '../../../utils/Constants';
import { BlueColor, FontStyle } from '../../../utils/Styles';
import { Icon } from '../../atoms/Icon';

export type Props = {
  onChangeText: (text: string) => void
}

export const ChatSearchbar = React.memo((props: Props) => {
  const { onChangeText } = props;
  const [text, setText] = useState<string>('');

  const updateText = (text: string) => {
    setText(text);
    onChangeText && onChangeText(text);
  };

  return (
      <View style={styles.searchBar}>
        <Icon name='filter' width={50} height={15} fill={BlueColor.mainColor}/>
        <TextInput
          style={{
              fontFamily: FontStyle.regular,
              fontSize: 14,
              lineHeight: 20,
              textAlignVertical: 'center',
              color: '#000',
              backgroundColor: 'transparent',
            }}
          placeholder={'検索'}
          placeholderTextColor={THEME_COLORS.OTHERS.LIGHT_GRAY}
          value={text}
          onChangeText={updateText}
      />
      </View>
  );
});

const styles = StyleSheet.create({
  searchBar: {
    height: 30,
    width: SCREEN_WIDTH * 0.9,
    justifyContent: 'flex-start',
    alignItems:'center',
    flexDirection: 'row',
		backgroundColor: 'transparent',
    borderBottomWidth: 1,
    borderBottomColor: '#555',
    marginTop: 10,
    marginBottom: 12,
    marginLeft: SCREEN_WIDTH * 0.1 * 0.5,
  },
});