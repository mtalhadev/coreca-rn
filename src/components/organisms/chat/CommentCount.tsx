import React from 'react'
import { Pressable, Text, View, ViewStyle } from 'react-native'

import { FontStyle } from '../../../utils/Styles'
import { useTextTranslation } from '../../../fooks/useTextTranslation'
import { Icon } from '../../atoms/Icon'

export type CommentCountProps = {
    count: number
    style?: ViewStyle
    onPress?: () => void
}

export const CommentCount = React.memo((props: Partial<CommentCountProps>) => {
    let { count, style, onPress } = props

    const { t } = useTextTranslation()
    return (
        <View
            style={[
                {
                    flexDirection: 'row',
                },
                style,
            ]}>

            <Pressable
                style={[
                    {
                        flexDirection: 'row',
                    },
                ]}
                onPress={onPress}
            >
                <Icon name='comment' width={20} height={20} />

                <Text
                    ellipsizeMode={'middle'}
                    style={{
                        fontFamily: FontStyle.regular,
                        fontSize: 12,
                        lineHeight: 18,
                        marginLeft: 5
                    }}>
                    {count?.toString()}
                </Text>
            </Pressable>
        </View>

    )
})
