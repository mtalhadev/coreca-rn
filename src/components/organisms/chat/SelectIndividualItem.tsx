import React, { useState } from 'react';
import { StyleSheet, Text, View } from 'react-native';
import { AppButton } from '../../atoms/AppButton';
import { ImageIcon } from '../ImageIcon';
import { useTextTranslation } from '../../../fooks/useTextTranslation';
import { SCREEN_WIDTH, THEME_COLORS } from '../../../utils/Constants';
import { FontStyle } from '../../../utils/Styles';
import { WorkerUIType } from './chatGroupMembers/SelectIndividual';

export const ChatUserItem = React.memo((props: WorkerUIType) => {
	const { t } = useTextTranslation();
	const { workerId, name, imageUrl, company, onChat } = props;

	const iconSize = 50
	const _imageUri = (iconSize <= 30 ? props?.xsImageUrl : (iconSize <= 50 ? props?.sImageUrl : props?.imageUrl)) ?? props?.imageUrl
	const _imageColorHue = props?.imageColorHue


	return (
		<View style={[styles.container]}>
			<ImageIcon
				imageUri={_imageUri}
				imageColorHue={_imageColorHue}
				type={'worker'}
				size={40}
				style={{ width: 40, height: 40, marginRight: 10 }} 
				borderRadius={40}
				borderWidth={1}
				/>
			<View style={styles.row}>
				<View style={styles.content}>
					<Text style={styles.title}>{name}</Text>
					<Text style={styles.subtitle} numberOfLines={1} ellipsizeMode='tail'>
						{company?.name}
					</Text>
				</View>
				<AppButton
					style={{
						backgroundColor: "transparent",
						paddingHorizontal: 20,
					}}
					height={32}
					hasShadow={false}
					borderColor={THEME_COLORS.OTHERS.BORDER_COLOR2}
					borderWidth={1}
					textColor={THEME_COLORS.OTHERS.BLACK}
					title={t('admin:ToChat')}
					onPress={() => {
						if(onChat)
							onChat({
								workerId
							})
					}}
					/>

			</View>
		</View>
	);
});

const SUB_ITEM_WIDTH = SCREEN_WIDTH-20;
const styles = StyleSheet.create({
	container: {
		width: SUB_ITEM_WIDTH,
		height: 50,
		alignItems: 'center',
		borderRadius: 3,
		marginBottom: 5,
		flexDirection: "row",
		paddingBottom: 0
	},
	content: {
		height: 45,
		justifyContent: 'space-between',
	},
	row: {
		width: SUB_ITEM_WIDTH - 50,
		flexDirection:"row",
		justifyContent: 'space-between',
		alignItems: 'center',
	},

	title: {
		fontFamily: FontStyle.medium,
		fontSize: 15,
		color: '#000'
	},
	subtitle: {
		fontFamily: FontStyle.regular,
		fontSize: 13,
		color: '#666'
	},
});