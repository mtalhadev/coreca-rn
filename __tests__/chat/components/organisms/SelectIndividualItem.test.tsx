import { render, screen } from '@testing-library/react-native';
import React from 'react';
import { ChatUserItem } from '../../../../src/components/organisms/chat/SelectIndividualItem';
import { GreenColor } from '../../../../src/utils/Styles';

jest.mock('react-i18next', () => ({
  // this mock makes sure any components using the translate hook can use it without a warning being shown
  useTranslation: () => {
    return {
      t: (str:string) => str,
      i18n: {
        changeLanguage: () => new Promise(() => {}),
      },
    };
  },
  initReactI18next: {
    type: '3rdParty',
    init: () => {},
  }
}));


describe('SelectIndividualItem test', () => {

  beforeEach(() => {
    jest.resetAllMocks()
  })

  it('renders UI correctly', () => {

    render(<ChatUserItem 
	    workerId = '12345678'
      name = 'ABC'
      xsImageUrl = 'https://profile-logo-xs'
      sImageUrl = 'https://profile-logo-s'
      imageUrl = 'https://profile-logo'
      imageColorHue = {50}
      company={{
        name: 'ABCDEFG',
      }}
    />
    );
    expect(screen.getByText('ABC')).toBeVisible();
    expect(screen.getByText('ABCDEFG')).toBeVisible();
    const image = screen.getByTestId("ImageIcon-Image")
    expect(image.props.source.uri).toBe("https://profile-logo-s");
    expect(image.props.style.width).toBe(40);
    expect(image.props.style.height).toBe(40);
    const imageBorder = screen.getByTestId("ImageIcon-Border")
    expect(imageBorder.props.style.borderColor).toBe(GreenColor.subColor);
    expect(screen).toMatchSnapshot();    

  });

})