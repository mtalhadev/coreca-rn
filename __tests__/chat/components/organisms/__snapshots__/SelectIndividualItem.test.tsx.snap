// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectIndividualItem test renders UI correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 3,
        "flexDirection": "row",
        "height": 50,
        "marginBottom": 5,
        "paddingBottom": 0,
        "width": 730,
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#888",
        "borderBottomLeftRadius": 40,
        "borderBottomRightRadius": 40,
        "borderRadius": 40,
        "borderTopLeftRadius": 40,
        "borderTopRightRadius": 40,
        "height": 40,
        "justifyContent": "center",
        "marginRight": 10,
        "width": 40,
      }
    }
  >
    <RNCMaskedView>
      <View
        pointerEvents="none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#000",
              "borderRadius": 40,
              "height": 40,
              "width": 40,
            }
          }
        />
      </View>
      <Image
        resizeMode="cover"
        source={
          Object {
            "uri": "https://profile-logo-s",
          }
        }
        style={
          Object {
            "height": 40,
            "width": 40,
          }
        }
        testID="ImageIcon-Image"
      />
    </RNCMaskedView>
    <View
      style={
        Object {
          "borderColor": "#C7D601",
          "borderRadius": 40,
          "borderWidth": 1,
          "height": 40,
          "position": "absolute",
          "width": 40,
        }
      }
      testID="ImageIcon-Border"
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": 680,
      }
    }
  >
    <View
      style={
        Object {
          "height": 45,
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#000",
            "fontFamily": "NotoSansJP_500Medium",
            "fontSize": 15,
          }
        }
      >
        ABC
      </Text>
      <Text
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "color": "#666",
            "fontFamily": "NotoSansJP_400Regular",
            "fontSize": 13,
          }
        }
      >
        ABCDEFG
      </Text>
    </View>
    <View
      collapsable={false}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "backgroundColor": "transparent",
          "borderColor": "#606060",
          "borderRadius": 32,
          "borderWidth": 1,
          "elevation": 0,
          "height": 32,
          "justifyContent": "center",
          "opacity": 1,
          "paddingHorizontal": 20,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 8,
        }
      }
    >
      <View
        accessibilityState={
          Object {
            "disabled": false,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": undefined,
              "paddingHorizontal": undefined,
            }
          }
        >
          <View
            style={
              Object {
                "alignContent": undefined,
                "flexDirection": undefined,
                "justifyContent": undefined,
                "paddingTop": 0,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#343538",
                  "fontFamily": "NotoSansJP_500Medium",
                  "fontSize": 14,
                  "lineHeight": 32,
                  "paddingTop": 0,
                  "textAlign": undefined,
                }
              }
            >
              admin:ToChat
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
